docs/runtimes/create-and-use-functions.ipynb

```diff
@@ -0,0 +1,554 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e790f68c",
+   "metadata": {},
+   "source": [
+    "This is a great way to create a function out of a Jupyter notebook  - simply pass in the path to the Jupyter notebook  *relative to your project context directory*. You can use [MLRun cell tags](#) to specify which parts of the notebook should be included in the function. \n",
+    "> Note: To ensure that the latest changes are included, make sure you save your notebook before creating/updating the function"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2e988b79",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "notebook_fn = single_source_project.set_function(\n",
+    "    name=\"notebook\", \n",
+    "    func=\"nb.ipynb\", \n",
+    "    kind=\"job\", \n",
+    "    image=\"mlrun/mlrun\", \n",
+    "    handler=\"handler\"\n",
+    ")\n",
+    "\n",
+    "run = single_source_project.run_function(notebook_fn)\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e10decbb",
+   "metadata": {},
+   "source": [
+    "#### Current Notebook"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e770b254",
+   "metadata": {},
+   "source": [
+    "You can also create an MLRun function out of the current Jupyter notebook you are running in. To do this, simply ommit the `func` parameter in `set_function`. You can use [MLRun cell tags](#) to specify which parts of the notebook should be included in the function. \n",
+    "> Note: To ensure that the latest changes are included, make sure you save your notebook before creating/updating the function"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "26f3d35b",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "notebook_fn = single_source_project.set_function(\n",
+    "    name=\"notebook\", \n",
+    "    kind=\"job\", \n",
+    "    image=\"mlrun/mlrun\", \n",
+    "    handler=\"handler\"\n",
+    ")\n",
+    "\n",
+    "run = single_source_project.run_function(notebook_fn)\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "84cf9258",
+   "metadata": {},
+   "source": [
+    "<a id=\"multiple-source-files\"></a>\n",
+    "### Multiple Source Files (Git/Archive)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "773d7bef",
+   "metadata": {},
+   "source": [
+    "If your code requires additional files or external libraries, you will need to use a source that supports multiple files such as Git, an archive (zip/tar/etc.), or V3IO file share. This approach (especially using a Git repo) pairs well with MLRun Projects.\n",
+    "\n",
+    "With this approach, you must provide `with_repo=True` into `set_function`. This instructs MLRun to load source code from the git repo/archive/file share associated with the project.\n",
+    "\n",
+    "There are two options to load the code:\n",
+    "1. **Static**: The function is built once. *This is the preferred approach for production workloads*\n",
```

Comment:
"- **Static**: The function is built once. *This is the preferred approach for production workloads*\n",
    "> Build the function once via `mlrun.build_function(fn)`\n",
    "- **Runtime**: The function pulls the source code at runtime. *This is is a simpler approach during development that allows for making code changes without re-building the image each time.*\n",
    "> Enable this feature via `project.set_source(source=None, pull_at_runtime=True)`"